kafka_acl_enforcer/src/main/java/com/tesla/data/acl/AclEnforceCommand.java

```diff
@@ -32,9 +34,15 @@ public AclEnforceCommand(Map<String, Object> cmdConfig, String cluster) {
   }
 
   @Override
-  protected Enforcer<AclBinding> initEnforcer(AdminClient client) {
+  protected Enforcer<AclBinding> initEnforcer() {
+    this.adminClient = KafkaAdminClient.create(kafkaConfig());
     return new AclEnforcer(configuredEntities(AclBinding.class, "acls", "aclsFile"),
-        new AclService(client), !unsafemode, dryrun);
+        new AclService(adminClient), !unsafemode, dryrun);
   }
-}
 
+  @Override
+  protected void close() {
+    assert this.adminClient != null;
```

Comment:
I'd change this to an if condition as opposed to an assert. Also, `assert` is not guaranteed to work, they are disabled by default https://docs.oracle.com/cd/E19683-01/806-7930/6jgp65ikq/index.html

Reply:
updated